File: Sources/UserMoodStatus.swift
import Foundation

struct UserMoodStatus: Identifiable {
    enum PrivacySetting: String, Codable {
        case allConnections = "ALL_CONNECTIONS"
        case selectedConnections = "SELECTED_CONNECTIONS"
        case `private` = "PRIVATE"
    }

    struct InferredSentiment: Codable {
        let score: Double
        let emotion: String
    }

    let id: UUID
    let userId: UUID
    let moodieType: String
    let actualScore: Int?
    let timestamp: Date
    let optionalContextText: String?
    var privacySetting: PrivacySetting = .allConnections
    let inferredSentiment: InferredSentiment?
}

File: models/userMoodStatus.js
class UserMoodStatus {
    constructor(id, userId, moodieType, actualScore, timestamp, optionalContextText, privacySetting, inferredSentiment) {
        this.id = id;
        this.userId = userId;
        this.moodieType = moodieType;
        this.actualScore = actualScore;
        this.timestamp = timestamp;
        this.optionalContextText = optionalContextText;
        this.privacySetting = privacySetting;
        this.inferredSentiment = inferredSentiment;
    }
}

module.exports = UserMoodStatus;
